iml-agent/src/action_plugins/ldev.rs

```diff
@@ -0,0 +1,325 @@
+// Copyright (c) 2020 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::{agent_error::ImlAgentError, env};
+use iml_wire_types::LdevEntry;
+use std::collections::BTreeSet;
+use tokio::{
+    fs::{File, OpenOptions},
+    io::{AsyncReadExt, AsyncWriteExt},
+};
+
+async fn write_to_file(content: String) -> Result<(), ImlAgentError> {
+    let ldev_path = env::get_ldev_conf();
+    let mut file = File::create(ldev_path).await?;
+    file.write_all(content.as_bytes()).await?;
+
+    Ok(())
+}
+
+async fn read_ldev_config() -> Result<String, ImlAgentError> {
+    let ldev_path = env::get_ldev_conf();
+
+    let mut file = OpenOptions::new()
+        .read(true)
+        .write(true)
+        .create(true)
+        .open(ldev_path)
+        .await?;
+
+    let mut buffer = String::new();
+    file.read_to_string(&mut buffer).await?;
+
+    Ok(buffer)
+}
+
+fn parse_entries(ldev_config: String) -> Result<BTreeSet<LdevEntry>, ImlAgentError> {
+    let existing_entries = ldev_config
+        .lines()
+        .map(LdevEntry::from)
+        .collect::<BTreeSet<LdevEntry>>();
+
+    Ok(existing_entries)
+}
+
+fn config_needs_update_check(
```

Comment:
Maybe just compare two strings: the one got from the file and the other rendered from the entries? :)

Reply:
If the order is different they won't match. I think the BTreeSet is more reliable.

Reply:
I think the order of ldev.conf doesn't matter, so we could just sort.